pkg/compose/compose_test: TestComposeCompare failed

pkg/compose/compose_test.TestComposeCompare [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/20795813?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_ComposeTestsBazel/20795813?buildTab=artifacts#/) on master @ [62084c233ab0355bfb5b87b172b363ad70333c9f](https://github.com/cockroachdb/cockroach/commits/62084c233ab0355bfb5b87b172b363ad70333c9f):


```
        cockroach2-1  | * This feature will be removed in a later version of CockroachDB.
        cockroach2-1  | *
        cockroach1-1  | CockroachDB node starting at 2025-12-02 06:16:25.043962939 +0000 UTC m=+1.406397918 (took 1.0s)
        cockroach1-1  | build:               CCL v26.1.0-alpha.1-dev-62084c233ab0355bfb5b87b172b363ad70333c9f @ 2025/12/02 06:06:57 (go1.25.3)
        cockroach1-1  | malloc_conf:         prof:false,prof_active:true,background_thread:true,thp:never,metadata_thp:disabled,narenas:32,dirty_decay_ms:2000,muzzy_decay_ms:0
        cockroach1-1  | webui:               http://cockroach1:8080
        cockroach1-1  | sql:                 postgresql://root@cockroach1:26257/defaultdb?sslmode=disable
        cockroach1-1  | sql (JDBC):          jdbc:postgresql://cockroach1:26257/defaultdb?sslmode=disable&user=root
        cockroach1-1  | RPC client flags:    /cockroach/cockroach <client cmd> --host=cockroach1:26257 --insecure
        cockroach1-1  | logs:                /cockroach/store/logs
        cockroach1-1  | temp dir:            /cockroach/store/cockroach-temp3181886693
        cockroach1-1  | external I/O path:   /cockroach/store/extern
        cockroach1-1  | store[0]:            path=/cockroach/store
        cockroach1-1  | clusterID:           2865c28e-9360-438c-b81e-921d9ff6c9e6
        cockroach1-1  | status:              initialized new cluster
        cockroach1-1  | nodeID:              1
        test-1        |     compare_test.go:114: Checking connection to: cockroach2
        cockroach2-1  | CockroachDB node starting at 2025-12-02 06:16:25.088227898 +0000 UTC m=+1.450670349 (took 1.0s)
        cockroach2-1  | build:               CCL v26.1.0-alpha.1-dev-62084c233ab0355bfb5b87b172b363ad70333c9f @ 2025/12/02 06:06:57 (go1.25.3)
        cockroach2-1  | malloc_conf:         prof:false,prof_active:true,background_thread:true,thp:never,metadata_thp:disabled,narenas:32,dirty_decay_ms:2000,muzzy_decay_ms:0
        cockroach2-1  | webui:               http://cockroach2:8080
        cockroach2-1  | sql:                 postgresql://root@cockroach2:26257/defaultdb?sslmode=disable
        cockroach2-1  | sql (JDBC):          jdbc:postgresql://cockroach2:26257/defaultdb?sslmode=disable&user=root
        cockroach2-1  | RPC client flags:    /cockroach/cockroach <client cmd> --host=cockroach2:26257 --insecure
        cockroach2-1  | logs:                /cockroach/store/logs
        cockroach2-1  | temp dir:            /cockroach/store/cockroach-temp2050632190
        cockroach2-1  | external I/O path:   /cockroach/store/extern
        cockroach2-1  | store[0]:            path=/cockroach/store
        cockroach2-1  | clusterID:           f3eb95dc-7fa9-4fbc-8b65-f57b0fff281e
        cockroach2-1  | status:              initialized new cluster
        cockroach2-1  | nodeID:              1
        test-1        | === RUN   TestCompare/mutators
        test-1        |     compare_test.go:123: starting test: mutators
        test-1        |     compare_test.go:132: initializing connection: cockroach1
        test-1        |     compare_test.go:132: initializing connection: cockroach2
        test-1        |     compare_test.go:162: pq: Access to crdb_internal and system is restricted.
        test-1        | --- FAIL: TestCompare (1.96s)
        test-1        |     --- FAIL: TestCompare/mutators (0.79s)
        test-1        | FAIL
        
�[Ktest-1 exited with code 1
        Aborting on container exit...
         Container compare-test-1  Stopping
         Container compare-test-1  Stopped
         Container compare-cockroach2-1  Stopping
         Container compare-cockroach1-1  Stopping
         Container compare-cockroach2-1  Stopped
         Container compare-cockroach1-1  Stopped
        
    compose_test.go:160: exit status 1
--- FAIL: TestComposeCompare (9.83s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestComposeCompare.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-57378

--- a/pkg/compose/compare/docker-compose.yml
+++ b/pkg/compose/compare/docker-compose.yml
@@ -5,6 +5,8 @@ services:
     image: us-east1-docker.pkg.dev/crl-docker-sync/docker-io/library/ubuntu:xenial-20170214
     user: ${UID}:${GID}
     command: /cockroach/cockroach start-single-node --insecure --store=/cockroach/store --spatial-libs=/cockroach/lib --listen-addr cockroach1
+    environment:
+      - COCKROACH_OVERRIDE_ALLOW_UNSAFE_INTERNALS=true
     volumes:
       - "${COCKROACH_PATH}:/cockroach/cockroach"
       - "${LIBGEOS_DIR_PATH}:/cockroach/lib"
@@ -16,6 +18,8 @@ services:
     # See https://cockroachlabs.atlassian.net/wiki/spaces/devinf/pages/3462594561/Docker+image+sync for the details.
     image: us-east1-docker.pkg.dev/crl-docker-sync/docker-io/library/ubuntu:xenial-20170214
     user: ${UID}:${GID}
+    environment:
+      - COCKROACH_OVERRIDE_ALLOW_UNSAFE_INTERNALS=true
     command: /cockroach/cockroach start-single-node --insecure --store=/cockroach/store --spatial-libs=/cockroach/lib --listen-addr cockroach2
     volumes:
       - "${COCKROACH_PATH}:/cockroach/cockroach"

